tests/bson/bson-fromPHP-006.phpt

```diff
@@ -0,0 +1,50 @@
+--TEST--
+BSON\fromPHP(): PHP documents with null bytes in field name
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+echo "\nTesting array with one leading null byte in field name\n";
+hex_dump(fromPHP(["\0" => 1]));
```

Comment:
While implementing this test, I reviewed [field name restrictions](https://docs.mongodb.com/manual/core/document/#field-names) for MongoDB documents, which prohibits `"$"` and `"."` characters in addition to null bytes. That said, the [BSON specification](http://bsonspec.org/spec.html) only prohibits null bytes on account of using C strings to encode field names, so I see no reason to enforce the other restrictions here. Validation of documents destined for MongoDB storage falls under [PHPC-712](https://jira.mongodb.org/browse/PHPC-712).

I'm also comfortable relying on `strlen()` to truncate the field name here instead of throwing an exception because the user included null bytes within a field name; however, I'd be happy to reconsider that. I expect it'd simply be a matter of comparing the string length reported by PHP's HashTable API with that of `strlen()` and throwing if there is a discrepancy.